plugin/trino-hive/src/main/java/io/trino/plugin/hive/crypto/EnvironmentDecryptionKeyRetriever.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.crypto;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import io.trino.parquet.crypto.DecryptionKeyRetriever;
+import org.apache.parquet.hadoop.metadata.ColumnPath;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.parquet.Preconditions.checkArgument;
+
+/**
+ * Reads keys from two environment variables.
+ * <pre>
+ *   pme.environment-key-retriever.footer-keys  =  &lt;single‑key&gt; | id1:key1,id2:key2 …
+ *   pme.environment-key-retriever.column-keys  =  &lt;single‑key&gt; | id1:key1,id2:key2 …
+ * </pre>
```

Comment:
Is this way of configuring based on something in Apache Hive or Apache Parquet ? If yes, would be good to add links to those in code comment

Reply:
ping